website/source/api/acl/role-binding-rules.html.md

```diff
@@ -0,0 +1,371 @@
+---
+layout: api
+page_title: ACL Role Binding Rules - HTTP API
+sidebar_current: api-acl-role-binding-rules
+description: |-
+  The /acl/rolebindingrule endpoints manage Consul's ACL Role Binding Rules.
+---
+
+-> **1.5.0+:**  The APIs are available in Consul versions 1.5.0 and later. The documentation for the legacy ACL API is [here](/api/acl/legacy.html)
+
+# ACL Role Binding Rule HTTP API
+
+The `/acl/rolebindingrule` endpoints [create](#create-a-role-binding-rule),
+[read](#read-a-role-binding-rule), [update](#update-a-role-binding-rule),
+[list](#list-roles) and [delete](#delete-a-role-binding-rule)  ACL role binding
+rules in Consul.  For more information about ACLs, please see the 
+[ACL Guide](/docs/guides/acl.html).
+
+## Create a Role Binding Rule
+
+This endpoint creates a new ACL role binding rule.
+
+| Method | Path                         | Produces                   |
+| ------ | ---------------------------- | -------------------------- |
+| `PUT`  | `/acl/rolebindingrule`       | `application/json`         |
+
+The table below shows this endpoint's support for
+[blocking queries](/api/index.html#blocking-queries),
+[consistency modes](/api/index.html#consistency-modes),
+[agent caching](/api/index.html#agent-caching), and
+[required ACLs](/api/index.html#acls).
+
+| Blocking Queries | Consistency Modes | Agent Caching | ACL Required |
+| ---------------- | ----------------- | ------------- | ------------ |
+| `NO`             | `none`            | `none`        | `acl:write`  |
+
+### Parameters
+
+- `Description` `(string: "")` - Free form human readable description of the role binding rule.
+
+- `IDPName` `(string: <required>)` - The name of the identity provider that
+  this rule applies to. This field is immutable.
+
+- `RoleName` `(string: <required>)` - The name of a role to bind to. Can
+  optionally use `{{ field_name }}` template syntax to generate a name using available
+  identity provider fields. Available fields are documented in the
+  [Identity Provider Guide](/docs/guides/acl-identity-providers.html).
+
+- `MustExist` `(bool: false)` - If true, indicates that at login time the named
+  role must already exist for this role binding rule to apply. This is described
+  in more detail in the [Identity Provider Guide](/docs/guides/acl-identity-providers.html).
+
+- `Matches` `(array<Match>)` - The list of match selectors. Individual matches
```

Comment:
**Warning** Possible user confusion.
Are we overloading terms here? Does a match selector here mean the same as with filtering? If not should we use different terms? @mkeeler

Reply:
Same goes for `selector`

Reply:
If @mkeeler thinks his new stuff is general purpose enough I could take a stab at rebasing the role binding rule syntax to use his new selector syntax instead.